flict/flictlib/project/reader.py

```diff
@@ -48,71 +48,95 @@ def _add_files_read(self, project_file):
         package_list = []
 
-        for _package in packages.values():
+        for _package in packages:
             tmp_dict = {}
-
-            for dep in _package['dependencies']:
+            for dep in _package.get('dependencies', []):
                 tmp_dict.update(self._flatten_package_tree(dep))
-
             dep_list = tmp_dict.values()
 
+            for dep in dep_list:
+                dep['dependencies'] = []
+
             package = {
                 'name': _package['name'],
                 'version': _package['version'],
                 'license': _package['license'],
                 'description': _package['description'],
-                'dependencies': dep_list,
+                'dependencies': list(dep_list),
             }
 
             package_list.append(package)
-        return (package_list)
+
+        return package_list
 
 
 class ProjectReaderFactory:
 
     @staticmethod
-    def get_projectreader(project_file=None, project_dirs=None, project_format=None):
+    def get_projectreader(project_file=None, project_dirs=None, project_format=None, update_dual=True):
         if not project_dirs:
             project_dirs = ['.']
-        logging.debug(f'get_projectreader({project_file}, {project_dirs}, {project_format})')
+        logging.debug(f'get_projectreader({project_file}, {project_dirs}, {project_format}, {update_dual})')
         if project_format is None:
             if 'spdx' in project_file:
-                return SPDXJsonProjectReader(project_dirs)
+                return SPDXJsonProjectReader(project_dirs, update_dual)
             if 'flict' in project_file:
-                return FlictProjectReader(project_dirs)
+                return FlictProjectReader(project_dirs, update_dual)
         elif project_format == 'flict':
-            return FlictProjectReader(project_dirs)
+            return FlictProjectReader(project_dirs, update_dual)
         elif project_format == 'spdx':
-            return SPDXJsonProjectReader(project_dirs)
+            return SPDXJsonProjectReader(project_dirs, update_dual)
 
 
 class FlictProjectReader(ProjectReader):
     """Class for reading flict project files"""
 
-    def __init__(self, project_dirs):
+    def __init__(self, project_dirs, update_dual=True):
+        self.update_dual = update_dual
         pass
 
+    def __read_project_data(self, _project_data):
+        project = _project_data['project']
+        project_data = {
+            'project_name': project['name'],
+            'packages': [
+                {
+                    'name': project['name'],
+                    'version': project['version'],
+                    'license': project['license'],
+                    'description': '',
+                    'dependencies': project['dependencies'],
+                },
+            ],
+        }
+        fixed_project_data = self.prepare_project(project_data)
+        return fixed_project_data
+
+    def read_project_data(self, project_data):
```

Comment:
This function seems to be used

Reply:
It is used, in `arbiter.py`

???

Reply:
right, I was just confused why we need a public and a private implementation, that are both the same (public is just an alias/forward for the private one) - maybe for better readability we can make the private implementation the public one